### ISSUE TITLE
no-unused-exports rule shows an false error when using destructured array exports: `exported declaration 'undefined' not used within other modules`

### ISSUE THREAD

**Issue by emreboga** [import-js/eslint-plugin-import#2930]
This issue is result of an incomplete fix for #2037.
Reposting my comment [here](https://github.com/import-js/eslint-plugin-import/pull/2038#issuecomment-1814121501) for visibility:

I don't think the issue was fully fixed by https://github.com/import-js/eslint-plugin-import/pull/1997 as it only addresses object type destucturing, not including array types. You can see it in the code:
https://github.com/import-js/eslint-plugin-import/pull/1997/files#diff-7d2908b010ec724c0f3a0d7615cb91f4673d765d52c6ba6607501e23470f2966R85

I'm seeing this in the code with the version 2.28.1, which has the fix with https://github.com/import-js/eslint-plugin-import/pull/1997. Here is a code chunk I have:
```
export const [authUserLoading, authUserError] = createLoadingAndErrorSelectors(
  AUTH_USER
);
```

and here is the error I get:

`error  exported declaration 'undefined' not used within other modules  import/no-unused-modules`

**Comment by ljharb** [comment]
Can you confirm that you *are* using it in other modules? meaning, is the issue that the error message is wrong, or that the warning itself is wrong?
